File: backend/src/main/java/net/tape/model/Author.java
package net.tape.model;

import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;

/** Domain model — framework-free POJO. */
@Getter
@Setter
public class Author implements Serializable {
    private Long id;
    private String name;
    private String bio;
    private Long avatarMediaId;
    private String imageUrl;
}

File: web/src/api/components/schemas/Author.ts
import { Option } from 'fp-ts/lib/Option';
import { Int, string, type } from 'io-ts';
import { optionFromNullable } from 'io-ts-types/lib/optionFromNullable';

export type Author = {
	id: Option<Int>;
	name: Option<string>;
	bio: Option<string>;
	avatarMediaId: Option<Int>;
	imageUrl: Option<string>;
};
export const AuthorIO = type(
	{
		id: optionFromNullable(Int),
		name: optionFromNullable(string),
		bio: optionFromNullable(string),
		avatarMediaId: optionFromNullable(Int),
		imageUrl: optionFromNullable(string),
	},
	'Author',
);
